remote-workstations/new-agent-vm/Install-PCoIPAgent.ps1

```diff
@@ -519,40 +519,62 @@ Configuration InstallPCoIPAgent
-                    -ScriptBlock {
-                        $domainGroupToJoin = $args[0]
-                        $machineToJoin = $args[1]
+                        $invokeResult = Invoke-Command -Session $psSession -ArgumentList $domainGroupToJoin, $machineToJoin  -ErrorAction Stop  `
+                        -ScriptBlock {
+                            $domainGroupToJoin = $args[0]
+                            $machineToJoin = $args[1]
 
-                        # Make the AD group for machines if needed
-                        try
-                        {
-                            Get-ADGroup $domainGroupToJoin -ErrorAction Stop
+                            # Make the AD group for machines if needed
+                            try
+                            {
+                                $group = Get-ADGroup -Filter {SamAccountName -eq $domainGroupToJoin} -ErrorAction Stop
+                                if ($group -eq $null) {
+                                    Write-Host "Domain Group `"$domainGroupToJoin`" not found. Creating."
+
+                                    New-ADGroup -name $domainGroupToJoin -GroupScope Global -ErrorAction Stop
+                                }
+
+                                # Add-ADGroupMember uses the SAM account name for the computer which has a trailing '$'
+                                Add-ADGroupMember -Identity $domainGroupToJoin -Members ($machineToJoin + "$") -ErrorAction Stop
+
+                                return $null
+                            }
+                            catch
+                            {
+                                return $_
+                            }
                         }
-                        catch
-                        {
-                            Write-Host "Domain Group `"$domainGroupToJoin`" not found. Creating."
 
-                            New-ADGroup -name $domainGroupToJoin -GroupScope Global
+                        if ($invokeResult -ne $null) {
+                            throw $invokeResult
                         }
+                    }catch{
+                        $exception = $_
 
-                        # Add-ADGroupMember uses the SAM account name for the computer which has a trailing '$'
-                        Add-ADGroupMember -Identity $domainGroupToJoin -Members ($machineToJoin + "$")
+                        Write-Host "Failed to add $machineToJoin to $domainGroupToJoin because of $exception."
+                    }finally{
+                        if ($psSession -ne $null) {
+                            Remove-PSSession $psSession
+                        }
                     }
-                    Remove-PSSession $psSession
                 }
                     
                 #make placeholder file so this is only run once
                 New-Item "$using:agentInstallerDLDirectory\domainGroupJoinFile.txt" -type file
+
+                if ($exception -ne $null) {
```

Comment:
Does this mean the DSC configuration doesn't complete if the domain join group failed?

Reply:
No. DSC still complete after join group failed because it still generate the flag file. It catch the exception and write the exception information to file

Reply:
Ah okay. And you're just writing the exception information to the file for debugging later?

Reply:
yes, you are right. Only exception information to the file for debugging. if joining group successful, then the flag file is empty as before.